### PR TITLE
Documentation of Client/HttpClient and data binding unit tests

### PR DESCRIPTION
**Pull request by Krukov**
#113 + #343

### CODE REVIEW

File: docs/generics.rst [hunk 1]
```diff
@@ -250,3 +250,40 @@ want to override; like so::
 You can also use the Django ``method_decorator`` utility to wrap methods that
 have ``message`` as their first positional argument - note that it won't work
 for more high-level methods, like ``WebsocketConsumer.receive``.
+
+
+As route
+--------
+
+Instead of make routes by ``route_class`` you may use ``as_route`` shortcut.
+This function take route filters (:ref:`filters`) as kwargs and return
```
Comment (ssteinerx) [inline docs/generics.rst]: This function takes

Comment (Krukov) [inline docs/generics.rst]: Thanks!

File: docs/generics.rst [hunk 2]
```diff
@@ -256,7 +256,7 @@ As route
 --------
 
 Instead of make routes by ``route_class`` you may use ``as_route`` shortcut.
-This function take route filters (:ref:`filters`) as kwargs and return
+This function takes route filters (:ref:`filters`) as kwargs and return
```
Comment (ssteinerx) [inline docs/generics.rst]: and returns

Comment (ssteinerx) [inline docs/generics.rst]: and returns

File: docs/generics.rst [hunk 3]
```diff
@@ -250,3 +250,40 @@ want to override; like so::
 You can also use the Django ``method_decorator`` utility to wrap methods that
 have ``message`` as their first positional argument - note that it won't work
```
Comment (ssteinerx) [inline docs/generics.rst]: argument.  Note that

File: docs/generics.rst [hunk 4]
```diff
@@ -250,3 +250,40 @@ want to override; like so::
...
 for more high-level methods, like ``WebsocketConsumer.receive``.
+
+
+As route
+--------
+
+Instead of make routes by ``route_class`` you may use ``as_route`` shortcut.
```
Comment (ssteinerx) [inline docs/generics.rst]: Instead of making routes using ``route_class`` you may use the ``as_route`` shortcut.

File: docs/generics.rst [hunk 5]
```diff
@@ -250,3 +250,40 @@ want to override; like so::
...
+the generic consumer::
+
+    class MyGenericConsumer(WebsocketConsumer):
+        group = 'default'
+        group_prefix = ''
+
+        def connection_groups(self, **kwargs):
+            return ['_'.join(self.group_prefix, self.group)]
+
+You can create consumers with differents ``group`` and  ``group_prefix`` with ``attrs``,
```
Comment (ssteinerx) [inline docs/generics.rst]: with different

File: docs/testing.rst [hunk 6]
```diff
@@ -108,6 +108,182 @@ do group adds and sends during a test. For example::
             self.assertEqual(result['value'], 42)
 
 
+Clients
+-------
+
+For more complicated test suites you can use ``Client`` abstraction that
```
Comment (ssteinerx) [inline docs/testing.rst]: can use the ``Client``

File: docs/testing.rst [hunk 7]
```diff
@@ -108,6 +108,182 @@ do group adds and sends during a test. For example::
...
+provides easy way to test full life cycle of messages with a couple of methods:
```
Comment (ssteinerx) [inline docs/testing.rst]: provides an easy way to test the full life cycle

**Comment by Krukov** [comment]
@ssteinerx Thank you for review!

File: docs/generics.rst (hunk 3 above)
Comment (Krukov) [inline docs/generics.rst]: This is not part of this PR. I can change it but only if @andrewgodwin will agree.

Comment (andrewgodwin) [inline docs/generics.rst]: I prefer the hyphen personally - it's really a stylistic thing at some point.

**Comment by andrewgodwin** [comment]
Looks good, thanks! I'll do a bit of copy-editing cleanup later this week but nothing major, just a few extra words here and there.

**Comment by ssteinerx** [comment]
@Krukov Quite welcome!

**Pull request merged**